exterminate.rb

```diff
@@ -167,24 +170,30 @@
 
     race_nr = df.world.raws.creatures.all.index { |cr| cr.creature_id == raw_race }
 
-    if caste
+    if caste == "enemy"
+        caste_nr = true
+    elsif caste
         all_castes = df.world.raws.creatures.all[race_nr].caste.map { |c| c.caste_id }
         raw_caste = df.match_rawname(caste, all_castes)
         if not raw_caste
             puts "Invalid caste, use one of #{all_castes.sort.join(' ')}"
             throw :script_finished
         end
         caste_nr = all_castes.index(raw_caste)
+    else
+        caste_nr = false
     end
 
     count = 0
-    df.world.units.active.each { |u|
+    df.world.units.active.each do |u|
         if u.race == race_nr and checkunit[u]
-            next if caste_nr and u.caste != caste_nr
+            if caste_nr
+                next if caste == "enemy" && df.unit_hostile(u)
```

Comment:
Should this be `!df.unit_hostile(u)`?

Reply:
It should. This is what I get for editing in the browser.